marklogic-client-api/src/main/java/com/marklogic/client/dataservices/CallBatcher.java

```diff
@@ -276,5 +279,11 @@
          * @return  a CallBatcher that accepts the input and makes calls with the caller
          */
         CallBatcher<CallManager.CallArgs,E> forArgs();
+        
+        CallBatcher<CallArgs,E> forArgsGenerator(CallArgsGenerator<E> generator);
```

Comment:
The technical story claims that forArgsGenerator() should return

`CallBatcher<Void,E>`

on the grounds that add() should not be supported as a way of providing input.